perfmetrics/scripts/ls_metrics/listing_benchmark.py

```diff
@@ -0,0 +1,473 @@
+optionally upload the results of the test to a Google Sheet. It takes input a
+JSON config file whcih contains the info regrading directory structure and also through
+which multiple tests of different configurations can be performed in a single run.
+
+Typical usage example:
+  $ python3 listing_benchmark.py [-h] [--keep_files] [--upload] [--num_samples NUM_SAMPLES] [--message MESSAGE] --command COMMAND config_file
+
+  Flag -h: Typical help interface of the script.
+  Flag --keep_files: Do not delete the generated directory structure from the persistent disk after running the tests.
+  Flag --upload: Uploads the results of the test to the Google Sheet.
+  Flag --num_smaples NUM_SAMPLES: Runs each test for NUM_SAMPLES times.
+  Flag --message MESSAGE: Takes input a message string, which describes/titles the test.
+  Flag --command COMMAND (required): Takes a input a string, which is the command to run the tests on.
+  config_file (required): Path to the JSON config file which contains the details of the tests.
+
+Note: This python script is dependent on generate_files.py.
+"""
+
+import argparse
+import ast
+import configparser
+import json
+import logging
+import os
+import statistics as stat
+import subprocess
+import sys
+import time
+
+import directory_pb2 as directory_proto
+
+sys.path.insert(0, '..')
+import generate_files
+import numpy as np
+import texttable
+
+from google.protobuf.json_format import Parse, ParseDict
+
+
+logging.basicConfig(
+    level=logging.INFO,
+    format='%(asctime)s [%(levelname)s] %(message)s',
+    handlers=[logging.StreamHandler(sys.stdout)],
+)
+log = logging.getLogger()
+
+
+def OutputResults(folders, results_list, message, num_samples) -> dict:
+  """Outputs the results on the console.
+
+  This function takes in dictionary containing the list of results (for all samples) for each
+  testing folder, for both the gcs bucket and persistent disk. Then it generates various metrics
+  out of these lists and outputs them into the console.
+
+  The metrics present in the output are (in msec):
+  Mean, Median, Standard Dev, 0th %ile, 20th %ile, 40th %ile, 60th %ile, 80th %ile, 90th %ile,
+  95th %ile, 98th %ile, 99th %ile, 100th %ile.
+
+  Args:
+    folders: List containing protobufs of testing folders.
+    results_list: Dictionary containing the list of results (for all samples)
+                  for each testing folder.
+    message: String which describes/titles the test.
+    num_samples: Number of samples to collect for each test.
+
+  Returns:
+    A dictionary containing the various metrics in a JSON format.
+  """
+  metrics = dict()
+
+  for testing_folder in folders:
+    metrics[testing_folder.name] = dict()
+
+    metrics[testing_folder.name]['Test Desc.'] = message
+    metrics[testing_folder.name]['Number of samples'] = num_samples
+    results_list[testing_folder.name] = sorted(results_list[testing_folder.name])
```

Comment:
is this to sort it based on time? can you add a comment.

Reply:
yes. added.